contrib/platform/src/com/sun/jna/platform/win32/WinCrypt.java

```diff
@@ -72,10 +72,15 @@ public DATA_BLOB(Pointer memory) {
         }
 
         public DATA_BLOB(byte [] data) {
-            pbData = new Memory(data.length);
-            pbData.write(0, data, 0, data.length);
+            if (data.length > 0) {
```

Comment:
You need a call to `super()` before this.

Reply:
no problem. is it to conform code convention?

Reply:
It's needed to allocate the memory for the structure, among other things.  We removed that memory allocation at the end, which seemed to imply we didn't have all the information needed until after we'd populated the pointer -- but we did, it's a fixed size array.

Reply:
But with my suggested solution the `getPointer()` would return null until it was allocated.

Reply:
I thought that javac adds `super()` call automatically.
https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.8.7
> If a constructor body does not begin with an explicit constructor invocation and the constructor being declared is not part of the primordial class Object, then the constructor body implicitly begins with a superclass constructor invocation "super();", an invocation of the constructor of its direct superclass that takes no arguments.

Reply:
OK, you're absolutely correct.  This is one of those cases where I've done it so much I forgot that it was just a "good idea".   So yes, I suppose it boils down to style, but here are some good reasons I choose to always do so here:
 - Consistency.  All the other constructors in this class have it (including the no-arg one).  This is common in JNA, where if one constructor declares it the no-arg one is almost always included as well (often the pair `super()` on the no-arg and `super(p); read();` on the pointer constructor) even if technically optional.  Omitting it looks like an oversight because it's "different".
 - Clarity.  Calling a no-arg superconstructor from a constructor with an argument is not common, and this makes it more clear that it's an explicit choice and not an oversight.
 - IDE navigation.  I often find myself double-checking just exactly what the `Structure` constructors do and it's nice to navigate directly to the correct line from within my IDE.
 - Error-checking.  Ever forgotten `extends` on the class when mapping JNA structures?  I have, more than once. (Don't laugh.). This helps catch that silly error early.

There may be other reasons for or against, particularly in the more general case, but for this particular class i think the "consistency" argument above is reason enough.